### ISSUE TITLE
Instant search (filter list)

### ISSUE THREAD

**Issue by paoloantinori** [ruyadorno/git-bc#4]
Hi, this addon is very handy. It would be even better if I could press a key like `/` and type a keyword and have the options filtered based on that.

**Comment by ruyadorno** [comment]
Yeah, it's a great idea, @zeachco might also be interested in following since he proposed the same thing :smile: 

The trickiest part is how to plug the filtering functionality along with [Inquirer.js](https://github.com/SBoudrias/Inquirer.js) (the **node** dependency that provides the interactive menu).

I'd be happy to review a contribution for that and might give it a try in the future if nobody else comes up with a PR first.

Thanks!

**Comment by zeachco** [comment]
Looked at it and I think inquirer is already working on an auto-complete feature (https://github.com/SBoudrias/Inquirer.js/issues/225) I also feel it would be a ore logical place to have it

**Comment by paoloantinori** [comment]
slowly getting there:

https://asciinema.org/a/2abz5r1f4sfsahhbl6161nc3f

**Comment by ruyadorno** [comment]
:+1: looking good!

**Comment by mokkabonna** [comment]
FYI I have a open PR for this now, feedback is wanted. https://github.com/SBoudrias/Inquirer.js/pull/265
